From 578ebcef8ba11a1dbf24d88b1e864007783e1f6d Mon Sep 17 00:00:00 2001
From: Viacheslav Bocharov <adeep@lexina.in>
Date: Fri, 18 Oct 2024 08:09:21 +0300
Subject: [PATCH] backport fix for ncurses.h include from commit
 263fe04f9ed6f3f40fdb7ce249221e03dc926c9b by Fabrice Fontaine
 <fontaine.fabrice@gmail.com>

Signed-off-by: Viacheslav Bocharov <adeep@lexina.in>
---
 configure.ac | 2 +-
 src/watch.c  | 7 +++++--
 2 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/configure.ac b/configure.ac
index fec27e3f..ed5dd47d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -64,7 +64,7 @@ PROCPS_PROG_PO4A
 
 # Checks for header files.
 AC_HEADER_MAJOR
-AC_CHECK_HEADERS([arpa/inet.h fcntl.h float.h langinfo.h libintl.h limits.h locale.h stdint.h stdio_ext.h stdlib.h string.h sys/file.h sys/ioctl.h sys/param.h sys/time.h termios.h unistd.h utmp.h utmpx.h values.h wchar.h wctype.h])
+AC_CHECK_HEADERS([arpa/inet.h fcntl.h float.h langinfo.h libintl.h limits.h locale.h ncursesw/ncurses.h stdint.h stdio_ext.h stdlib.h string.h sys/file.h sys/ioctl.h sys/param.h sys/time.h termios.h unistd.h utmp.h utmpx.h values.h wchar.h wctype.h])
 
 # Checks for typedefs, structures, and compiler characteristics.
 AC_CHECK_HEADER_STDBOOL
diff --git a/src/watch.c b/src/watch.c
index 5c159a95..d968baca 100644
--- a/src/watch.c
+++ b/src/watch.c
@@ -34,7 +34,6 @@
 #include "nls.h"
 #include "strutils.h"
 #include "xalloc.h"
-#include <ctype.h>
 #include <errno.h>
 #include <getopt.h>
 #include <locale.h>
@@ -53,10 +52,14 @@
 # define _XOPEN_SOURCE_EXTENDED 1
 # include <wchar.h>
 # include <wctype.h>
+#else
+#include <ctype.h>
+#endif	/* WITH_WATCH8BIT */
+#ifdef HAVE_NCURSESW_NCURSES_H
 # include <ncursesw/ncurses.h>
 #else
 # include <ncurses.h>
-#endif	/* WITH_WATCH8BIT */
+#endif
 
 #ifdef FORCE_8BIT
 # undef isprint
-- 
2.47.0

